src/words/environment/WordsEventListener.java

```diff
@@ -1,15 +1,35 @@
 package words.environment;
 
 import words.ast.AST;
+import words.ast.ASTValue;
+import words.environment.WordsEnvironment;
+import words.exceptions.WordsRuntimeException;
 
 public class WordsEventListener {
-	
 	private AST predicate;
 	private AST statementList;
+	private boolean temporary;
+
+	public WordsEventListener(AST predicate, AST statementList, boolean temporary) {
+		this.predicate = predicate;
+		this.statementList = statementList;
+		this.temporary = temporary;
+	}
 	
-	public void execute() {
-		// TODO Auto-generated method stub
-		
+	public boolean execute(WordsEnvironment environment) throws WordsRuntimeException {
```

Comment:
I think that this should probably throw a WordsProgramException, since this is where the AST lives.